src/code128.ps.src

```diff
@@ -396,6 +398,171 @@ begin
+        /prioritized_latch_length_b1 [reverse_priority {dup dup latch_length_b1 exch get exch} forall] def
+        /prioritized_latch_length_c0 [reverse_priority {dup dup latch_length_c0 exch get exch} forall] def
+        /prioritized_latch_length_c1 [reverse_priority {dup dup latch_length_c1 exch get exch} forall] def
+
+        /max_int 2147483647 def    % Make sure state doesn't get picked.
+
+        % Predicates for ability to encode:
+        /can_a  {c          0 ge {true} {seta c known} ifelse} def
+        /can_b  {c          0 ge {true} {setb c known} ifelse} def
+        /can_c0 {num_digits 2 ge {true} {setc c known} ifelse} def
+        /can_c1 {num_digits 2 ge {true} {setc c known} ifelse} def
+
+        % Predicates overruled by options:
+        suppressc                      {/can_c0 {false} def} if
+        suppressc unlatchextbeforec or {/can_c1 {false} def} if
+
+        % Output length:
+        /out_a0 {1 c 0 ge {c 128 ge {1 add} if c 127 and 96 ge {1 add} if} if} def
+        /out_a1 {1 c 0 ge {c 128 lt {1 add} if c 127 and 96 ge {1 add} if} if} def
+        /out_b0 {1 c 0 ge {c 128 ge {1 add} if c 127 and 32 lt {1 add} if} if} def
+        /out_b1 {1 c 0 ge {c 128 lt {1 add} if c 127 and 32 lt {1 add} if} if} def
+
+        % Encode:
+        /map_ab {dup 32 lt {64 add} {32 sub} ifelse} def
+        /enc_a0 {[c 0 lt {seta c get} {c 128 ge {101} if c 127 and dup 96 ge {98 exch} if map_ab} ifelse]} def
+        /enc_a1 {[c 0 lt {seta c get} {c 128 lt {101} if c 127 and dup 96 ge {98 exch} if map_ab} ifelse]} def
+        /enc_b0 {[c 0 lt {setb c get} {c 128 ge {100} if c 127 and dup 32 lt {98 exch} if map_ab} ifelse]} def
+        /enc_b1 {[c 0 lt {setb c get} {c 128 lt {100} if c 127 and dup 32 lt {98 exch} if map_ab} ifelse]} def
+        /enc_c  {[c 0 lt {setc c get} {msg n get 48 sub 10 mul msg n 1 add get 48 sub add       } ifelse]} def
+
+        % Get best prior state based on a prior row of lengths and a row of latch
+        % lengths (unrolled and with preprocessed latch lengths on the stack):
+        /get_best_prior_state {
+            bln_0 exch get add             /len exch def /o exch def
+            bln_0 exch get add dup len lt {/len exch def /o exch def} {pop pop} ifelse
+            bln_0 exch get add dup len lt {/len exch def /o exch def} {pop pop} ifelse
+            bln_0 exch get add dup len lt {/len exch def /o exch def} {pop pop} ifelse
+            bln_0 exch get add dup len lt {/len exch def /o exch def} {pop pop} ifelse
+            bln_0 exch get add     len lt {              /o exch def} {    pop} ifelse
+            o
+        } def
+
+        % The encoder considers the current row and two rows back.
+        % The circular history buffer size is 4 for convenience.
+        % The names are short to keep the lines below reasonable:
+        /bln_0 start_length def /bln_1 start_length def /bln [4 {[0 0 0 0 0 0]} repeat] def % Best Length
```

Comment:
Better to have double spaces before comments: `... repeat] def %  Best Length`

(So that it doesn't read like a modulus operator from other languages.)